pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/DateFilter.java

```diff
@@ -88,19 +103,36 @@ private Object applyDate(Object dateOrString, final PebbleTemplate self, final L
     if (timeZone != null) {
       intendedFormat.setTimeZone(TimeZone.getTimeZone(timeZone));
     }
+
+    if (alterTimeDuration != null) {
+      date = Date.from(date.toInstant().plus(alterTimeDuration));
+    }
+
     return new SafeString(intendedFormat.format(date));
   }
 
-  private Object applyTemporal(final TemporalAccessor input, PebbleTemplate self,
+  private Object applyTemporal(Temporal input, PebbleTemplate self,
       final Locale locale,
-      int lineNumber, final String format, final String timeZone) throws PebbleException {
+      int lineNumber, final String format, final String timeZone,
+      Duration alterTimeDuration) throws PebbleException {
     DateTimeFormatter formatter = format != null
         ? DateTimeFormatter.ofPattern(format, locale)
         : DateTimeFormatter.ISO_DATE_TIME;
 
     ZoneId zoneId = getZoneId(input, timeZone);
     formatter = formatter.withZone(zoneId);
 
+    if (alterTimeDuration != null) {
+      try {
+        input = input.plus(alterTimeDuration);
```

Comment:
How would we handle negative time offests?

Reply:
It all works out, `-P1D` is supported.

Reply:
I could write some test for it to make it clear. It will have to go into the docs anyway though.